### PR TITLE
[JBPM-9986] Smart router is sending requested to stopped server in RHPAM 7

### PR DESCRIPTION
**Pull request by elguardian**
Jira https://issues.redhat.com/browse/JBPM-9986

### CODE REVIEW

**Comment by elguardian** [comment]
race conditon cannot be guarded as it is per design.

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Review by gmunozfe — COMMENTED** [review]
Looks good to me, astounding work @elguardian !
Some comments about the tests and also a possible former bug (when alias and containerId are the same) if you consider fixing it here.

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java [hunk 1]
```diff
@@ -267,28 +291,102 @@ public void onConfigurationReloaded() {
...
+        configurationManager.close();
+        executorService.shutdownNow();
+        repoWithWatcher.close();
+        repoWithWatcher.clean();
+    }
+
+    @Test
+    public void testConfigurationFileStartStop() throws Exception {
+
+        String fileDir = "target" + File.separator + UUID.randomUUID().toString();
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java:322]: done

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java [hunk 2]
```diff
@@ -267,28 +291,102 @@ public void onConfigurationReloaded() {
...
+
+        CountDownLatch latch = new CountDownLatch(1);
+        loaded.addListener(new ConfigurationListener() {
+
+            @Override
+            public void onConfigurationReloaded() {
+                latch.countDown();
+            }
 
+        });
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java:342]: done

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java [hunk 3]
```diff
@@ -267,28 +291,102 @@ public void onConfigurationReloaded() {
...
+        loaded.addContainerInfo(containerInfo);
+
+        containerId = "container2";
+        containerInfo = new ContainerInfo(containerId, containerId, releaseId);
+        loaded.addContainerHost(containerId, serverUrl);
+        loaded.addServerHost(serverId, serverUrl);
+        loaded.addContainerInfo(containerInfo);
+
+        repoWithWatcher.persist(loaded);
+        System.out.println(loaded);
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java:375]: done

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java [hunk 4]
```diff
@@ -267,28 +291,102 @@ public void onConfigurationReloaded() {
...
+        loaded.removeServerHost(serverId, serverUrl);
+        loaded.removeContainerInfo(containerInfo);
+
+        containerId = "container1";
+        loaded.removeContainerHost(containerId, serverUrl);
+        loaded.removeServerHost(serverId, serverUrl);
+        loaded.removeContainerInfo(containerInfo);
+        repoWithWatcher.persist(loaded);
+
+        System.out.println(loaded);
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java:387]: correct.. just check the server2 hosts

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/Configuration.java [hunk 5]
```diff
@@ -38,7 +40,15 @@
...
-    
+
+    private List<ResponseAggregator> aggregators = new ArrayList<>();
+
+    public Configuration() {
+        this.aggregators.add(new JSONResponseAggregator());
+        this.aggregators.add(new XstreamXMLResponseAggregator());
+        this.aggregators.add(new JaxbXMLResponseAggregator());
+    }
+
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/Configuration.java:51]: done +test

**Pull request merged**